zio-http/src/main/scala/zhttp/http/Http.scala

```diff
@@ -512,6 +521,41 @@ object Http {
    */
   def html(view: Html): HttpApp[Any, Nothing] = Http.response(Response.html(view))
 
+  /**
+   * Creates an HTTP app to serve static resource files from a local directory.
+   */
+  def fromPath(root: jPath): HttpApp[Any, Nothing] = {
+    def contentTypeHeader(file: java.io.File): Header   =
+      (HttpHeaderNames.CONTENT_TYPE, Files.probeContentType(file.toPath))
+    def contentLengthHeader(file: java.io.File): Header = (HttpHeaderNames.CONTENT_LENGTH, file.length().toString)
+
+    def responseHttp(file: File) = responseZIO {
+      for {
+        data <- Task(HttpData.fromFile(file))
+        res  <- ZIO.succeed(Response(data = data))
+      } yield res
+    } @@ updateHeaders(_.addHeaders(Headers(Chunk(contentTypeHeader(file), contentLengthHeader(file)))))
+
+    Http.collectHttp[Request] { case request =>
+      if (request.method != Method.GET)
+        Http.methodNotAllowed(s"${request.method} is not allowed here. Please use `GET` instead.")
+      else {
+        val file = Paths.get(root.toString + "/" + request.path.asString).toFile
+        if (file.isDirectory)
+          response(Response(data = HttpData.fromString(listFilesHtml(file.toPath))))
+        else
+          responseHttp(file)
+      }.catchAll {
+        case a: SecurityException     =>
+          Http.error(HttpError.Forbidden(a.getMessage))
```

Comment:
```suggestion
          Http.forbidden(a.getMessage)
```

I am not sure if we should use `getMessage`, security is one thing but the other thing is that the value could be null.